realm/realm-library/src/androidTestObjectServer/java/io/realm/SchemaTests.java

```diff
@@ -145,4 +147,54 @@ public void allow_addField() {
 
         realm.close();
     }
+
+    @Test
+    public void addPrimaryKey_notAllowed() {
+        String className = "StringOnly";
+        Realm realm = Realm.getInstance(config);
+
+        realm.beginTransaction();
+        RealmObjectSchema objectSchema = realm.getSchema().get(className);
+        objectSchema.addField("foo", String.class);
+
+        try {
+            objectSchema.addPrimaryKey("foo");
```

Comment:
Yup, we also agreeded that we should keep this for non-synced Realms. While we generally want to keep the API as similar as possible, it doesn't seem possible for migrations where synced Realms have a lot of other restrictions as well.